File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/fragments/createnew/state/CreateNewStateImpl.java
package ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.createnew.state;

import java.util.List;

import ua.meugen.android.pdfcomposer.model.data.PageContent;
import ua.meugen.android.pdfcomposer.model.utils.CollectionUtils;
import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.state.BaseState;


public class CreateNewStateImpl extends BaseState implements CreateNewState {

    private static final String PARAM_PAGES = "pages";

    @Override
    public List<PageContent> getPages() {
        return bundle.getParcelableArrayList(PARAM_PAGES);
    }

    @Override
    public void setPages(final List<PageContent> pages) {
        bundle.putParcelableArrayList(PARAM_PAGES, CollectionUtils.toArrayList(pages));
    }
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/fragments/createnew/state/CreateNewStateImpl.kt
package ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.createnew.state

import ua.meugen.android.pdfcomposer.model.data.PageContent
import ua.meugen.android.pdfcomposer.model.utils.toArrayList
import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.state.BaseState


class CreateNewStateImpl : BaseState(), CreateNewState {

    override var pages: List<PageContent>?
        get() = bundle!!.getParcelableArrayList(PARAM_PAGES)
        set(pages) = bundle!!.putParcelableArrayList(PARAM_PAGES, pages!!.toArrayList)

    companion object {

        private const val PARAM_PAGES = "pages"
    }
}
